WordPress/create-block-theme
# Process inner html of blocks when escaping text content

**What**
This PR enables HTML markup inside block content to be processed while still allowing the text content to be safely escaped.

Before | After
--- | ---
<video src="https://github.com/user-attachments/assets/a9c1a925-5622-43cc-ace3-5a2942f5b936"> | <video src="https://github.com/user-attachments/assets/367f866e-3f9e-4cc5-b781-0b336e19dc5f">

**How**
Updates the `CBT_Theme_Locale::escape_text_content` function to handle nested HTML markup. 

Uses `WP_HTML_Tag_Processor` to process every token of the blocks content, and generates a string that can be formatted and then translated. 

**Why**
Fixes #573 #682 and #691.  

**To test**
- Add some markup to the inside of a block in a template.
- Save the changes to the theme.
- Verify the template has been patternized correctly, and the template appears as expected.
